### ISSUE TITLE
Annoying pedestrians

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#77]
```
I am in Fallhaven. The walkways are mostly one one tile wide. There are many 
pedestrians. Problem: a pedestrian can block the way - there is no way around. 
I have to wait until he hast moved to a spot where I can bypass him. As the 
pedestrians are moved randomly that can be VERY annoying.

The same with the drunkards in the inn. The often block the way (ok - in that 
case it seems realistic...).

I would like an option, which allows me to "jump" over pedestrians. So I can 
walk past somebody without to have to wait for him or to chat with him.
```

Original issue reported on code.google.com by `erwr...@gmail.com` on 10 Dec 2010 at 8:45

**Comment by GoogleCodeExporter** [comment]
```
My suggestions:
1. Maybe add a "Chat" button, which will only appear if you are next to NPC 
(non-player character)? If there is more then one NPC next to you, there could 
be a popup (for example: "Talk to Drunk" "Talk to Ganos") or something like 
that.
2. By default, you chat with NPC if you try to move in his/her direction. Maybe 
set an "auto switch position" as a default action for those situations? Of 
course, there could be exceptions (important NPC's, which don't move).

Sorry for my english :) 
```

Original comment by `rafal.wa...@gmail.com` on 10 Dec 2010 at 11:25

**Comment by GoogleCodeExporter** [comment]
```
1) The chat button is a good suggestion. (as an option)
The opening dialogs are very annoying in such cases.

Improvement:
Perhaps also a "push the NPC"-functionality. If I move towards him he will try 
to leave the tile possibly moving other NPCs if no adjacent tile exists.
```

Original comment by `SamuelPl...@gmail.com` on 11 Dec 2010 at 10:40

**Comment by GoogleCodeExporter** [comment]
```
Popup would be ok if I don't have to dismiss it. So just walk away and the 
popup disapears is ok. To have to tap on the popup to dismiss it, is much more 
bad then the actual thing. A chat-button which is strictly optional is a good 
idea. "pushing npc's" would be fun but is quite unrealistic.

I think the best solution would be that the hero can simply swap places with a 
npc but not with a monster or with a "key npc" (traders, story-tellers, 
quest-targets etc.). By the way: key npcs can become "normal" npc if the quest 
is closed.


```

Original comment by `erwr...@gmail.com` on 11 Dec 2010 at 12:30

**Comment by GoogleCodeExporter** [comment]
```
I assume the slightest influence on current UI would be adding a "pass by" 
button whenever there is a "bye/leave" button. e.g.
Trade - Leave - Pass by

This would work either by putting the char over npc (displayed by blinking the 
two characters on same tile), or switching positions.

In any case, some NPCs (guardians of passages/entrances, particularly, quest 
monsters :) ) may override this (i.e. no pass-by for them). E.g. clerics tend 
to stay close to their altars -- they cannot be moved (and no reason, really)
```

Original comment by `surrano` on 21 Dec 2010 at 2:54

**Comment by GoogleCodeExporter** [comment]
```
Sounds good. Definitly better then now. I would prefer switching positions 
instead of char over npc.
```

Original comment by `erwr...@gmail.com` on 1 Jan 2011 at 9:52

**Comment by GoogleCodeExporter** [comment]
Original comment by `sdeva...@gmail.com` on 10 Feb 2011 at 12:48
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Another much easier option would be to make the maps in such a way that 
walkways are more than 1 tile wide. 

Maybe we should update the Fallhaven maps to widen the walkways. But this is a 
really low priority issue in my opinion.
```

Original comment by `oskar.wi...@gmail.com` on 10 Feb 2011 at 7:31
- Added labels: **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
I think widening walkways may be the best option. However, I also found this 
very frustrating myself, because sometimes I'll see an NPC who just doesn't 
want to leave (mostly at the top-leftmost map)
```

Original comment by `jch...@email.wm.edu` on 8 Apr 2011 at 4:51

**Comment by GoogleCodeExporter** [comment]
```
Issue 178 has been merged into this issue.
```

Original comment by `sdeva...@gmail.com` on 19 Apr 2011 at 8:44

**Comment by GoogleCodeExporter** [comment]
```
Why not just have no collision on these characters,  with a button to chat as 
mentioned before
```

Original comment by `robindegen` on 28 Mar 2012 at 9:00

**Comment by GoogleCodeExporter** [comment]
```
Marking this as ready for v0.6.11. Walkways in Fallhaven have been widened, to 
remedy the situation. If we feel the need for some more advanced solution to 
pushing NPCs, then we'll open a new issue for that.
```

Original comment by `oskar.wi...@gmail.com` on 29 Apr 2012 at 3:52
- Changed state: **Ready**
- Added labels: **Milestone-0.6.11**

**Comment by GoogleCodeExporter** [comment]
```
OK. Is fine for me.
```

Original comment by `erwr...@gmail.com` on 30 Apr 2012 at 5:26

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.11. Closing issue.
```

Original comment by `oskar.wi...@gmail.com` on 9 Jun 2012 at 9:31
- Changed state: **Fixed**
